components/net/fetch/methods.rs

```diff
@@ -264,7 +264,15 @@ pub fn main_fetch(request: Rc<Request>,
         }
 
         // Step 16
-        // TODO this step (CSP/blocking)
+        // TODO Blocking for CSP, mixed content, MIME type
+        let blocked_error_response;
+        let internal_response = if should_block_nosniff(&request, &response) {
+            // Defer rebinding result
+            blocked_error_response = Response::network_error(NetworkError::Internal("Blocked by nosniff".into()));
+            &blocked_error_response
+        } else {
+            internal_response
+        };
 
         // Step 17
         if !response.is_network_error() && (is_null_body_status(&internal_response.status) ||
```

Comment:
You are correct. We should add a test for that, since currently it will pass the unit tests either way. I will see if I can add something.